src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -224,6 +225,13 @@ internal override bool TryGetSpeculativeSemanticModelCore(SyntaxTreeSemanticMode
 
             var methodSymbol = (MethodSymbol)this.MemberSymbol;
             binder = new WithNullableContextBinder(SyntaxTree, position, binder);
+
+            // PROTOTYPE(semi-auto-props): We should traverse until we get a property accessor, similar to Binder_Expressions. We may want to share the code.
+            if (methodSymbol is SourcePropertyAccessorSymbol { Property.IsIndexer: false } accessor)
+            {
+                binder = new SpeculativeFieldKeywordBinder(accessor, binder);
```

Comment:
>binder = new SpeculativeFieldKeywordBinder(accessor, binder); [](http://example.com/codeflow?start=16&length=61)

Do we have a test hitting this code path?

Reply:
Yes. `SpeculativeSemanticModel_FieldKeywordInRegularAccessor_ReplaceBlock_BindSpeculatedFirst` will fail if we removed this line.